test/has-route.test.js

```diff
@@ -71,7 +71,7 @@ test('hasRoute', t => {
 
     t.equal(fastify.hasRoute({
       method: 'GET',
-      url: '/example/12345.png'
+      url: '/example/:file(^\\d+).png'
```

Comment:
Ay any reason for the change?

Reply:
hasRoute is now a strict matcher. Before it was checking if one route is matching the pattern.